rpm2archive.c

```diff
@@ -187,34 +187,34 @@ static int process_package(rpmts ts, char * filename)
 
 int main(int argc, char *argv[])
 {
-    int rc;
-    
+    int rc, i;
+
     xsetprogname(argv[0]);	/* Portability call -- see system.h */
     rpmReadConfigFiles(NULL, NULL);
-    char * filename;
-    if (argc == 1)
-	filename = "-";
-    else {
-	if (rstreq(argv[1], "-h") || rstreq(argv[1], "--help")) {
-	    fprintf(stderr, "Usage: rpm2archive file.rpm\n");
-	    exit(EXIT_FAILURE);
-	} else {
-	    filename = argv[1];
-	}
-    }
 
-    rpmts ts = rpmtsCreate();
-    rpmVSFlags vsflags = 0;
-
-    /* XXX retain the ageless behavior of rpm2cpio */
-    vsflags |= RPMVSF_MASK_NODIGESTS;
-    vsflags |= RPMVSF_MASK_NOSIGNATURES;
-    vsflags |= RPMVSF_NOHDRCHK;
-    (void) rpmtsSetVSFlags(ts, vsflags);
-
-    rc = process_package(ts, filename);
+    if (argc > 1 && (rstreq(argv[1], "-h") || rstreq(argv[1], "--help"))) {
+	fprintf(stderr, "Usage: %s [file.rpm ...]\n", argv[0]);
+	exit(EXIT_FAILURE);
+    }
 
-    ts = rpmtsFree(ts);
+    if (argc == 1)
+	argv[argc++] = "-";
```

Comment:
This doesn't look right. You can't just write beyond the bounds of the argv array. You need to allocate a new - bigger - one.

Reply:
Per C standard, argv[argc] is always defined to be a NULL pointer.  (Section 5.1.2.2.1, paragraph 2, in ISO/IEC 9899:1999).

Reply:
Now you may not *like* that I'm doing this, which is a perfectly valid complaint and I'm happy to change it :)

Reply:
Yeah, we don't like that. Please change :)